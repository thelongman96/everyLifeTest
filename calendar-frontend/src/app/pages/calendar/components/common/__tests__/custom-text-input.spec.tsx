import { act, render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import CustomTextInput from '../custom-text-input';

const onChangeMock = jest.fn();
const onSubmitMock = jest.fn();

const defaultProps = {
  value: '',
  fieldId: 'test',
  fieldLabel: 'test',
  fieldType: '',
  fieldAutoComplete: '',
  fieldPlaceholder: '',
  errors: [],
  onChange: onChangeMock,
  onSubmit: onSubmitMock,
  fullWidth: false,
  customProps: {},
  sx: {},
};

const renderCustomTextInput = (props = defaultProps) => {
  return render(
    <CustomTextInput
      value={props.value}
      fieldId={props.fieldId}
      fieldLabel={props.fieldLabel}
      fieldType={props.fieldType}
      fieldAutoComplete={props.fieldAutoComplete}
      fieldPlaceholder={props.fieldPlaceholder}
      errors={props.errors}
      onChange={props.onChange}
      onSubmit={props.onSubmit}
      fullWidth={props.fullWidth}
      customProps={props.customProps}
      sx={props.sx}
    />,
  );
};

describe('CustomTextInputComponent', () => {
  it('renders correctly', () => {
    const { container } = renderCustomTextInput();

    expect(container).toMatchSnapshot();
  });

  describe('On Change - ', () => {
    it('should call onChange', async () => {
      renderCustomTextInput();

      const input = screen.getByTestId('test_input');
      await act(async () => {
        await fireEvent.change(input, { target: { value: 'test' } });
      });

      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
