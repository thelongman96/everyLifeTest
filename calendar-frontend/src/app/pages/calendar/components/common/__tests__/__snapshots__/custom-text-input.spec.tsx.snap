// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInputComponent renders correctly 1`] = `
<div>
  <div
    class="customTextInput__container"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="test"
        id="test-label"
      >
        test
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="test_input"
          id="test"
          name="test"
          placeholder=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1n64csd-MuiNotchedOutlined-root"
          >
            <span>
              test
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
